kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -499,6 +498,98 @@ private void genUnitAxiom(Production prod, StringBuilder sb) {
+            setArgsCeil = KApply(andLabel, setArgsCeil, KApply(ceilVarLabel, setVar));
+        }
+        Seq<K> argsSeq = JavaConverters.iterableAsScalaIterable(args).toSeq();
+        Seq<K> setArgsSeq = JavaConverters.iterableAsScalaIterable(setArgs).toSeq();
+        Rule inKeysRule1 = Rule(
+                KRewrite(
+                        KApply(prod.klabel().get(),
+                                args.get(0),
+                                KApply(concatProd.klabel().get(),
+                                        KApply(elementProd.klabel().get(),
+                                                argsSeq,
+                                                Att.empty()
+                                        ),
+                                        restMap
+                                )
+                        ),
+                        BooleanUtils.TRUE
+                ),
+                BooleanUtils.TRUE,
+                BooleanUtils.TRUE
+        );
+        rules.add(inKeysRule1);
+        Rule inKeysRule2 = Rule(
+                KRewrite(
+                        KApply(prod.klabel().get(),
+                                args.get(0),
+                                restMap
+                        ),
+                        BooleanUtils.FALSE
+                ),
+                BooleanUtils.TRUE,
+                BooleanUtils.TRUE,
+                Att.empty().add("owise")
+        );
+        rules.add(inKeysRule2);
+        KLabel equalsLabel = KLabel("#Equals", Sorts.Bool(), sortParam);
+        Rule ceilMapRule =
+                Rule(
+                        KRewrite(
+                                KApply(ceilMapLabel,
+                                        KApply(concatProd.klabel().get(),
+                                                KApply(elementProd.klabel().get(),
+                                                        setArgsSeq,
+                                                        Att.empty()
+                                                ),
+                                                restMapSet
+                                        )
+                                )
+                                ,
+                                KApply(andLabel,
+                                        KApply(equalsLabel,
+                                                KApply(prod.klabel().get(),
+                                                        setArgs.get(0),
+                                                        restMapSet
+                                                ),
+                                                BooleanUtils.FALSE
+                                        ),
+                                        setArgsCeil
+                                )
+                        )
+                        , BooleanUtils.TRUE
+                        , BooleanUtils.TRUE
+                        , Att.empty().add("anywhere").add("simplification")
```

Comment:
What is the `anywhere` for?

Reply:
That's awkward, but since #Ceil is not a function, we usually need to annotate `#Ceil` rules with `anywhere` to prevent them from being completed with `<k>` and stuff. Maybe here I didn't need to add them, as we're late in the compilation process, but I added them for consistency.